### ISSUE TITLE
How could I remove this warning: "Could not perform 'modify_collision_properties' on any prims under"?

### ISSUE THREAD

**Issue by waivivian** [isaac-sim/IsaacLab#3603]
2025-10-02T10:27:33Z [43,091ms] [Warning] [isaaclab.sim.utils] Could not perform 'modify_collision_properties' on any prims under: '/World/envs/env_59/Robot'. This might be because of the following reasons:
        (1) The desired attribute does not exist on any of the prims.
        (2) The desired attribute exists on an instanced prim.
                Discovered list of instanced prim paths: ['/World/envs/env_59/Robot/world/visuals', '/World/envs/env_59/Robot/world/collisions', '/World/envs/env_59/Robot/right_index1/visuals', '/World/envs/env_59/Robot/right_index1_y/visuals', '/World/envs/env_59/Robot/right_index1_z/visuals', '/World/envs/env_59/Robot/right_index1_z/collisions', '/World/envs/env_59/Robot/right_index2/visuals', '/World/envs/env_59/Robot/right_index2_y/visuals', '/World/envs/env_59/Robot/right_index2_z/visuals', '/World/envs/env_59/Robot/right_index2_z/collisions', '/World/envs/env_59/Robot/right_index3/visuals', '/World/envs/env_59/Robot/right_index3_y/visuals', '/World/envs/env_59/Robot/right_index3_z/visuals', '/World/envs/env_59/Robot/right_index3_z/collisions', '/World/envs/env_59/Robot/right_middle1/visuals', '/World/envs/env_59/Robot/right_middle1_y/visuals', '/World/envs/env_59/Robot/right_middle1_z/visuals', '/World/envs/env_59/Robot/right_middle1_z/collisions', '/World/envs/env_59/Robot/right_middle2/visuals', '/World/envs/env_59/Robot/right_middle2_y/visuals', '/World/envs/env_59/Robot/right_middle2_z/visuals', '/World/envs/env_59/Robot/right_middle2_z/collisions', '/World/envs/env_59/Robot/right_middle3/visuals', '/World/envs/env_59/Robot/right_middle3_y/visuals', '/World/envs/env_59/Robot/right_middle3_z/visuals', '/World/envs/env_59/Robot/right_middle3_z/collisions', '/World/envs/env_59/Robot/right_pinky1/visuals', '/World/envs/env_59/Robot/right_pinky1_y/visuals', '/World/envs/env_59/Robot/right_pinky1_z/visuals', '/World/envs/env_59/Robot/right_pinky1_z/collisions', '/World/envs/env_59/Robot/right_pinky2/visuals', '/World/envs/env_59/Robot/right_pinky2_y/visuals', '/World/envs/env_59/Robot/right_pinky2_z/visuals', '/World/envs/env_59/Robot/right_pinky2_z/collisions', '/World/envs/env_59/Robot/right_pinky3/visuals', '/World/envs/env_59/Robot/right_pinky3_y/visuals', '/World/envs/env_59/Robot/right_pinky3_z/visuals', '/World/envs/env_59/Robot/right_pinky3_z/collisions', '/World/envs/env_59/Robot/right_ring1/visuals', '/World/envs/env_59/Robot/right_ring1_y/visuals', '/World/envs/env_59/Robot/right_ring1_z/visuals', '/World/envs/env_59/Robot/right_ring1_z/collisions', '/World/envs/env_59/Robot/right_ring2/visuals', '/World/envs/env_59/Robot/right_ring2_y/visuals', '/World/envs/env_59/Robot/right_ring2_z/visuals', '/World/envs/env_59/Robot/right_ring2_z/collisions', '/World/envs/env_59/Robot/right_ring3/visuals', '/World/envs/env_59/Robot/right_ring3_y/visuals', '/World/envs/env_59/Robot/right_ring3_z/visuals', '/World/envs/env_59/Robot/right_ring3_z/collisions', '/World/envs/env_59/Robot/right_thumb1/visuals', '/World/envs/env_59/Robot/right_thumb1_y/visuals', '/World/envs/env_59/Robot/right_thumb1_z/visuals', '/World/envs/env_59/Robot/right_thumb1_z/collisions', '/World/envs/env_59/Robot/right_thumb2/visuals', '/World/envs/env_59/Robot/right_thumb2_y/visuals', '/World/envs/env_59/Robot/right_thumb2_z/visuals', '/World/envs/env_59/Robot/right_thumb2_z/collisions', '/World/envs/env_59/Robot/right_thumb3/visuals', '/World/envs/env_59/Robot/right_thumb3_y/visuals', '/World/envs/env_59/Robot/right_thumb3_z/visuals', '/World/envs/env_59/Robot/right_thumb3_z/collisions']

**Comment by PeterL-NV** [comment]
Hi, I don't think this warning is harmful. Do you have any issues with the converted USD?
Another way to convert your URDF to USD is to import it directly in Isaac Sim. Please check this [tutorial](https://docs.isaacsim.omniverse.nvidia.com/5.0.0/importer_exporter/import_urdf.html#)

**Comment by waivivian** [comment]
Thanks a lot!

Might I know why does this happen?

    scene: InteractiveSceneCfg = InteractiveSceneCfg(num_envs=6, env_spacing=1.0, replicate_physics=True)
    contact: ContactSensorCfg = ContactSensorCfg(
            prim_path="/World/envs/env_.*/Robot/.*",
            filter_prim_paths_expr=[f"/World/envs/env_.*/Object" ],
        )
2025-10-06T08:49:35Z [27,213ms] [Error] [omni.physx.tensors.plugin] Filter pattern '/World/envs/env_*/Object' did not match the correct number of entries (expected 276, found 6)

I have 6 envs and each env there are 46 bodies

**Comment by PeterL-NV** [comment]
The error happens because the filtering system in Isaac Lab contact sensors only supports "one-to-many" matching across environments. When you use regex patterns that match multiple sensor bodies and multiple filtered bodies together, the system expects that for every sensor body in every environment, there is a corresponding filtered body, leading to a required match count that may not align with the shapes of your expressions.
For filtering to work, your `prim_path` for `ContactSensorCfg` must match only a single body per environment

**Comment by waivivian** [comment]
Thanks a lot! However, I saw in a NVlabs project ProtoMotions code, it used ContactSensorCfg in a similar way to mine, as "/World/envs/env_.*/Robot/bodies/.*" should be referring to multiple bodies?             

            self.contact_sensor: ContactSensorCfg = ContactSensorCfg(
                prim_path="/World/envs/env_.*/Robot/.*",
                filter_prim_paths_expr=[f"/World/objects/object_{i}" for i in range(0)],
            )

<img width="953" height="703" alt="Image" src="https://github.com/user-attachments/assets/3a882cc2-5aa1-46a4-b72c-ad9384d00c1d" />

So how did it work for their code? Thanks in advance!

**Comment by PeterL-NV** [comment]
I don't know how many contact sensors are in their robot. But from the API doc on [ContactSensorCfg](https://isaac-sim.github.io/IsaacLab/main/source/api/lab/isaaclab.sensors.html#isaaclab.sensors.ContactSensorCfg), the sensor primitive [prim_path](https://isaac-sim.github.io/IsaacLab/main/source/api/lab/isaaclab.sensors.html#isaaclab.sensors.ContactSensorCfg.prim_path) corresponds to a single primitive in that environment. If the sensor primitive corresponds to multiple primitives, the filtering will not work as expected.

<img width="2700" height="802" alt="Image" src="https://github.com/user-attachments/assets/1b65353f-3bee-41f2-b7de-2e9a099e0a02" />

**Comment by waivivian** [comment]
https://github.com/isaac-sim/IsaacLab/issues/2588
Might I know why for this case, they can transverse all contact sensors by contact_sensor.data.net_contact_forces.norm() ?
I would also like to so this if possible as my current way of implementation seems to inefficient that I need to use for loop to loop over all contact sensor

 for body_name in self.cfg.keyBodies:
        sensor_name = f"contact_sensor_{body_name}"
        if sensor_name in self.scene.sensors:
            contact_data = self.scene.sensors[sensor_name].data
            
            # force_matrix_w shape: [num_envs, num_bodies, num_max_contacts, 3]
            contact_forces = contact_data.force_matrix_w.clone()
            print(contact_forces.shape)
            # Sum over contact points to get net force per body
            net_contact_forces = contact_forces.sum(dim=2)  # [num_envs, num_bodies, 3]
            
            # Calculate force magnitudes
            force_magnitudes = torch.norm(net_contact_forces, dim=2)  # [num_envs, num_bodies]
            
            # Check if any body has significant contact force in each environment
            contact_threshold = 0.1
            body_has_contact = torch.any(force_magnitudes > contact_threshold, dim=1)  # [num_envs]


    Thanks in advance!

**Comment by PeterL-NV** [comment]
Hi @waivivian, I don't what's your set up is, but seems what you need is a many body contact sensor, can you refer to this [Contact Sensor](https://isaac-sim.github.io/IsaacLab/main/source/overview/core-concepts/sensors/contact_sensor.html#contact-sensor) example to see if that helps?

**Comment by tesslerc** [comment]
@waivivian , regarding ProtoMotions, we're soon updating how we create contact sensors
```
for obj_idx in range(num_objects_per_scene):
    sensing_filter.append(f"/World/envs/env_.*/Object_{obj_idx}")
for body_name in robot_config.contact_bodies:
    contact_sensor_cfg = ContactSensorCfg(
        prim_path=f"{robot_config.asset.usd_bodies_root_prim_path}{body_name}",
        filter_prim_paths_expr=sensing_filter,
        history_length=config.sim.decimation
    )
    setattr(self, f"contact_sensor_{body_name}", contact_sensor_cfg)
```

You can try this approach.
This way you create a contact sensor for each body part that filters out multiple elements in the environment.
So 1-to-many is respected.

Then your code can do:
```
for body_name in self._body_names:
    if f"contact_sensor_{body_name}" in self._scene.keys():
        self._contact_sensor_map[body_name] = self._scene[f"contact_sensor_{body_name}"]
```
and then
```
    def _get_simulator_bodies_contact_buf(
        self, env_ids: Optional[torch.Tensor] = None
    ) -> RobotState:
        """
        Retrieve the contact force buffer for simulation bodies in sim body order.

        Parameters:
            env_ids (Optional[torch.Tensor]): Specific environments to query.

        Returns:
            RobotState: Robot state containing contact forces in simulator body order.
        """
        # Get simulator body ordering
        sim_body_names = self._robot.data.body_names
        num_bodies = len(sim_body_names)
        
        # Pre-allocate tensor for contact forces (initialized to zeros)
        rigid_body_contact_forces = torch.zeros(
            self.num_envs, num_bodies, 3, device=self.device
        )
        
        # Fill in contact forces for bodies that have sensors
        for body_idx, body_name in enumerate(sim_body_names):
            if body_name in self._contact_sensor_map:
                contact_sensor = self._contact_sensor_map[body_name]
                # net_forces_w has shape [num_envs, 1, 3], extract the single body dimension
                rigid_body_contact_forces[:, body_idx, :] = contact_sensor.data.net_forces_w.clone()[:, 0, :]
        
        if env_ids is not None:
            rigid_body_contact_forces = rigid_body_contact_forces[env_ids]
```

**Comment by waivivian** [comment]
Thanks a lot for both of your replies
